pyccel/stdlib/lists/lists.c

```diff
@@ -0,0 +1,244 @@
+#include "lists.h"
+
+
+t_list   *allocate_list(size_t size, t_type type, void *elemnts)
+{
+    t_list *list;
+    size_t tsize = tSizes[type];
+
+    if (!(list = (t_list*)malloc(sizeof(t_list))))
+        return NULL;
+    list->size = size;
+    list->type = type;
+    list->capacity = DEFAULT_CAP;
+    while (list->capacity <= list->size)
+        list->capacity *= 2;
+    if (!(list->elements = malloc(list->capacity * tsize)))
+        return NULL;
+    if (elemnts)
+        memcpy(list->elements, elemnts, tsize * list->size);
+    return list;
+}
+
+void     free_list(t_list **list)
+{
+    free((*list)->elements);
+    free(*list);
+    *list = NULL;
+}
+
+void    extend(t_list* list1, t_list* list2)
+{
+    size_t totalSize = list1->size + list2->size;
+    size_t tsize = tSizes[list1->type];
+    char *elements = list1->elements;
+
+    if (totalSize >= list1->capacity)
+    {
+        while (list1->capacity <= totalSize)
+            list1->capacity *= 2;
+        elements = realloc(elements, list1->capacity * tsize);
+        list1->elements = elements;
+    }
+    memcpy(&elements[list1->size], list2->elements, list2->size * tsize);
+    list1->size = totalSize;
+}
+
+void     clear(t_list* list)
+{
+    list->size = 0;
+}
+
+t_list*  copy(t_list* list)
+{
+    return (allocate_list(list->size, list->type, list->elements));
+}
+
+size_t   count(t_list* list, void *item)
+{
+    size_t index = 0;
+    int count = 0;
+    char *elements = list->elements;
+    while (index < list->size)
+    {
+        if (list->type == lst_list)
+        {
+            t_list *tmp = ((t_list**)elements)[index];
+            if (!(memcmp(item, tmp, sizeof(t_list))))
+                count += 1;
+        }
+        else if (!(memcmp(item, &(elements[index * tSizes[list->type]]), tSizes[list->type])))
+                count += 1;
+        index += 1;
+    }
+    return (count);
+}
+
+void     append(t_list* list, void* item)
+{
+    t_list listTmp;
+    listTmp.capacity = DEFAULT_CAP;
+    listTmp.elements = item;
+    listTmp.size = 1;
+    listTmp.type = list->type;
+    extend(list, &listTmp);
+}
```

Comment:
@azouiten We can simply call `insert` in this case.
This code is not viable because we create an instance if t_list on the stack, which makes unreachable once out of the stack frame of the function `append`. This will cause segmentation fault.